src/File/FileInformationTrait.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-validator for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-validator/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Validator\File;
+
+use Psr\Http\Message\UploadedFileInterface;
+use Zend\Validator\Exception;
+
+trait FileInformationTrait
+{
+    /**
+     * Returns array if the procedure is identified
+     *
+     * @param  string|array|object $value    Filename to check
+     * @param  array               $file     File data from \Zend\File\Transfer\Transfer (optional)
+     * @param  bool                $hasType  Return with filetype (optional)
+     * @param  bool                $basename Return with basename - is calculated from location path (optional)
+     * @return array
+     */
+    protected function getFileInfo(
+        $value,
+        $file = null,
+        $hasType = false,
+        $hasBasename = false
+    ) {
+        if (is_string($value) && is_array($file)) {
+            return $this->getLegacyFileInfo($file, $hasType, $hasBasename);
+        }
+
+        if (is_array($value)) {
+            return $this->getSapiFileInfo($value, $hasType, $hasBasename);
+        }
+
+        if ($value instanceof UploadedFileInterface) {
+            return $this->getPsr7FileInfo($value, $hasType, $hasBasename);
+        }
+
+        return $this->getFileBasedFileInfo($value, $hasType, $hasBasename);
+    }
+
+    /**
+     * Generate file information array with Legacy Zend\Transfer API
+     *
+     * @param object $file        File data from \Zend\File\Transfer\Transfer
```

Comment:
same here about `object` type

Reply:
This one MUST be an array; see the second argument to `getFileInfo()`.